server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java

```diff
@@ -487,6 +487,105 @@ public void testExplain() throws Exception {
             )
             .setRefreshPolicy(IMMEDIATE)
             .get();
+
+        SearchResponse searchResponse = client().prepareSearch("test")
+            .setQuery(nestedQuery("nested1", termQuery("nested1.n_field1", "n_value1"), ScoreMode.Total))
+            .setExplain(true)
+            .get();
+        assertNoFailures(searchResponse);
+        assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
+        Explanation explanation = searchResponse.getHits().getHits()[0].getExplanation();
+        assertThat(explanation.getValue(), equalTo(searchResponse.getHits().getHits()[0].getScore()));
+        assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs"));
+        assertTrue(explanation.toString().contains("in range from 0 to 1"));
+    }
+
+    public void testExplainMultipleDocIndexingNoDeletedDoc() throws Exception {
+        assertAcked(
+            prepareCreate("test").setMapping(
+                jsonBuilder().startObject()
+                    .startObject("properties")
+                    .startObject("nested0")
+                    .field("type", "nested")
+                    .endObject()
+                    .startObject("nested1")
+                    .field("type", "nested")
+                    .endObject()
+                    .startObject("nested2")
+                    .field("type", "nested")
+                    .endObject()
+                    .endObject()
+                    .endObject()
+            )
+        );
+
+        ensureGreen();
+
+        for (int i = 0; i < 3; i++) {
+            client().prepareIndex("test")
+                .setId(String.valueOf(i))
+                .setSource(
+                    jsonBuilder().startObject()
+                        .field("field" + i, "value" + i)
+                        .startArray("nested" + i)
+                        .startObject()
+                        .field("n_field" + i, "n_value" + i)
+                        .endObject()
+                        .startObject()
+                        .field("n_field" + i, "n_value" + i)
+                        .endObject()
+                        .endArray()
+                        .endObject()
+                )
+                .setRefreshPolicy(IMMEDIATE)
+                .get();
+        }
+
+        SearchResponse searchResponse = client().prepareSearch("test")
+            .setQuery(nestedQuery("nested1", termQuery("nested1.n_field1", "n_value1"), ScoreMode.Total))
+            .setExplain(true)
+            .get();
+        assertNoFailures(searchResponse);
+        assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
+        Explanation explanation = searchResponse.getHits().getHits()[0].getExplanation();
+        assertThat(explanation.getValue(), equalTo(searchResponse.getHits().getHits()[0].getScore()));
+        assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs"));
```

Comment:
Can we understand why range is changing with multiple docs even when data is constant. And can't we know the range before hand to add in assertion here ?

Also in deleted docs case, given the deleted docs are some garbage docs why is it affecting the explain output ?

Reply:
@sohami I did some investigation and found out that the range specifies the docIds of the child docs in the explain output. DocIds are loaded from the scoreDocs of the topDocs [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/SearchService.java#L1469).

After that, during the execute of the FetchPhase, call comes to the [ExplainPhase](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/fetch/subphase/ExplainPhase.java#L64) that passes docId to the explain function in lucene.
So, as an example say a document with doc Id 83 matches with the query passed, and there are 2 child docs that are part of the parent doc matching the query, the explain API will output contain `Score based on 2 child docs in range from 81 to 82, using score mode Total`.

In case of concurrent search case, since we are indexing so many documents, the docId is varying and hence the explain output contains the range for different numbers. Hope that answers your question.